source/slang/check.cpp

```diff
@@ -1681,7 +1681,7 @@ namespace Slang
         {
             // TODO: actually implement matching here. For now we'll
             // just pretend that things are satisfied in order to make progress..
-            requirementDict.Add(requiredMemberDeclRef, DeclRef<Decl>(memberDecl, nullptr));
+            requirementDict.AddIfNotExists(requiredMemberDeclRef, DeclRef<Decl>(memberDecl, nullptr));
```

Comment:
Why are we hitting the "key already exists" case here? Is this a case where we are checking the same requirement more than once, because it has been inherited along multiple paths?

Reply:
Yes, I’m the example code, we have extension IBxDF : IAcceptSphereLight, where IAcceptSphereLight:IBxDF. So IBxDF:IBxDF. And a decl can be added twice coming from different pathes.